src/lib/MapCards.svelte

```diff
@@ -62,6 +72,9 @@
             </div>
             <div class="tag-row-container">
               <Tags tags={map.tags} />
+              <div class="map-preview">
+                <MapPreview mapId={map.id} {setPreviewKey} />
```

Comment:
Adding a non-reusable component to render the button for this to which a `setPreviewKey` has to be passed seems unnecessarily complicated for no benefit. Without the component you could also get rid of the setPreviewKey entirely.

Reply:
It could potentially be reused for MOTW, no?

Reply:
Could be. It's a hypothetical as of now and it might be odd to have it on MotW since it undermines the Showcase.

But my biggest gripe with this is rather, that this button component has an internal state that it can decide to provide outside. Binding the key on this is also odd. But I did find this: https://svelte.dev/repl/acef5f41aae44264904acd4d240f3fa6?version=3.19.1 (imagine only one layer to it).

If it had a less confusing name and you'd just bind your on:click on this, making this a nice dumb component, I'd be pretty happy with that as well.